src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.ctre.phoenix.sensors.CANCoder;
+import com.ctre.phoenix.sensors.CANCoderConfiguration;
+import com.ctre.phoenix.sensors.SensorTimeBase;
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.SparkMaxAbsoluteEncoder;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.SubsystemChecker;
+import frc.robot.SubsystemChecker.SubsystemType;
+import frc.robot.config.Config;
+
+public class ArmSubsystem extends SubsystemBase {
+
+  private static final MotorType motorType = MotorType.kBrushless;
+  private static final SparkMaxAbsoluteEncoder.Type encAbsType = SparkMaxAbsoluteEncoder.Type.kDutyCycle;
+  
+  public CANCoder m_absoluteTopArmEncoder;
+  public CANCoder m_absoluteBottomArmEncoder;
+
+
+  private static ArmSubsystem instance = null;
+
+
+  public final CANSparkMax m_topArm;
+  public final CANSparkMax m_bottomArm;
+  public SparkMaxPIDController m_pidControllerTopArm;
+  public SparkMaxPIDController m_pidControllerBottomArm;
+  public double kP, kI, kD, kIz, kFF, kMaxOutput, kMinOutput;
+
+  /** Creates a new ArmSubsystem. */
+  public ArmSubsystem() {
+    m_topArm = new CANSparkMax(Config.CANID.TOP_ARM_SPARK_CAN_ID, motorType);
```

Comment:
This is a mighty long constructor. To make code easier to read, I recommend extracting things to helper methods, that are called inside the constructor. Functionally it's no different, but it helps readability a lot. A good examples would be "configure PID coefficients". In VSCode, you can highlight the code you want to extract to a method, right click -> refactor.. and select extract to method.